nixos/modules/services/misc/pinnwand.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pinnwand;
+
+  configFile = pkgs.runCommand "pinnwand.toml" {
+    buildInputs = [ pkgs.remarshal ];
+    preferLocalBuild = true;
+  } ''
+    remarshal -if json -of toml \
+    < ${pkgs.writeText "pinnwand.toml" (builtins.toJSON cfg.config)} \
+    > $out
+  '';
+in
+{
+  options.services.pinnwand = {
+    enable = mkEnableOption "Pinnwand";
+
+    port = mkOption {
+      type = types.int;
+      description = "The port to listen on";
+      default = 8000;
+    };
+
+    config = mkOption {
+      type = types.uniq types.attrs;
+      description = ''
+        Your <filename>pinnwand.toml</filename> as a Nix attribute set. Look up
+        possible options in the <link xlink:href="https://github.com/supakeen/pinnwand/blob/master/pinnwand.toml-example">pinnwand.toml-example</link>.
+      '';
+      default = {
+        # https://github.com/supakeen/pinnwand/blob/master/pinnwand.toml-example
+        database_uri = "sqlite:///var/lib/pinnwand/pinnwand.db";
```

Comment:
Does `pinwand` support other database types? Would we be forcing users to store credentials in the nix store under certain scenarios here? Would a separate set of `database` options provide value here?

Reply:
It does, I'm not sure else I could pass them to the application though.